libs/components/forms/src/lib/modules/checkbox/checkbox.component.ts

```diff
@@ -46,48 +47,62 @@ export class SkyCheckboxComponent implements ControlValueAccessor, OnInit {
    * If the checkbox does not include a visible label, use `label` instead.
    */
   @Input()
-  public labelledBy: string;
+  public labelledBy: string | undefined;
 
   /**
    * Specifies an ID for the checkbox.
    * @default a unique, auto-incrementing integer. For example: `sky-checkbox-1`
    */
+  @HostBinding('attr.id')
   @Input()
-  public id = `sky-checkbox-${++nextId}`;
+  public set id(value: string | undefined) {
+    if (value) {
+      this.#_id = value;
+      this.inputId = `input-sky-checkbox-${value}`;
+    } else {
+      const defaultId = `sky-checkbox-${++nextId}`;
+      this.#_id = defaultId;
+      this.inputId = `input-${defaultId}`;
+    }
+  }
+
+  public get id(): string {
+    return this.#_id;
+  }
 
   /**
    * Indicates whether to disable the checkbox.
    * @default false
    */
   @Input()
-  public set disabled(value: boolean) {
+  public set disabled(value: boolean | undefined) {
     const coercedValue = SkyFormsUtility.coerceBooleanProperty(value);
     if (coercedValue !== this.disabled) {
```

Comment:
We shouldn't reference the getter in the setter.

Reply:
Fixed